### PR TITLE
:bug: Source Recharge: added old api version for orders stream

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/oncall/issues/2833
User doesn't have email, first_name, last_name columns in orders stream, because API v2022-11 removed it

## How
Reverted to old API version 2022-01 where this fields are present

### CODE REVIEW

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py [hunk 1]
```diff
@@ -174,8 +174,14 @@ class Onetimes(IncrementalRechargeStream):
 class Orders(IncrementalRechargeStream):
     """
     Orders Stream: https://developer.rechargepayments.com/v1-shopify?python#list-orders
+    Using old API version to avoid schema changes and loosing email, first_name, last_name columns, because in new version it not present
     """
 
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {"x-recharge-version": OLD_API_VERSION}
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-recharge/source_recharge/api.py:183]: I think it's better to use a specific variable name format such as `API_VERSION_2021_01`. This approach will help us avoid confusion as we continue to introduce multiple versions of API for streams in the future.

**Comment by roman-yermilov-gl** [comment]
There is another option that let us not to rollback to old API version. We can add `?include=customer` query param: (https://developer.rechargepayments.com/2021-11/extending_responses), so we will have entier `customer` model included in response in this way:
```
{
    "id": 534919106,
    ...,
    "customer": {
        "id": 64962974,
        "email": "email@example.com",
        "external_customer_id": {
            "ecommerce": "5213433266347"
        },
        "hash": "f99bd4a6877257af"
    },
    "include": {
        "customer": {
            "id": 10000000,
            "analytics_data": {
                "utm_params": []
            },
            "created_at": "2021-05-13T09:46:44+00:00",
            "email": "email@example.com",
            "external_customer_id": {
                "ecommerce": "5213433266347"
            },
            "first_charge_processed_at": "2021-05-13T13:46:39+00:00",
            "first_name": "Banana",
            "has_payment_method_in_dunning": false,
            "has_valid_payment_method": true,
            "hash": "f99bd4a6877257af",
            "last_name": "Green",
            "phone": "+11111111111",
            "subscriptions_active_count": 1,
            "subscriptions_total_count": 1,
            "tax_exempt": false,
            "updated_at": "2023-05-13T04:16:36+00:00"
        }
    }
    ...
}
```
It let us transform a record, so we can put missing fields from `include.customer.<field>` to `customer.<field>`. There are also some disadvatages: record will contain more data and we will have some duplicated fields.
But advantages are: switching API version is a little bit destructive change because those versions are not backward compatible. So with this option we will break nothing but extend a model with some missing parts as well as keep API updated.
I am not requesting change. This is just another option of how we can fix this problem

**Review by lazebnyi — APPROVED** [review]

**Comment by darynaishchenko** [comment]
> There is another option that let us not to rollback to old API version. We can add `?include=customer` query param: (https://developer.rechargepayments.com/2021-11/extending_responses), so we will have entier `customer` model included in response in this way:
> 
> ```
> {
>     "id": 534919106,
>     ...,
>     "customer": {
>         "id": 64962974,
>         "email": "email@example.com",
>         "external_customer_id": {
>             "ecommerce": "5213433266347"
>         },
>         "hash": "f99bd4a6877257af"
>     },
>     "include": {
>         "customer": {
>             "id": 10000000,
>             "analytics_data": {
>                 "utm_params": []
>             },
>             "created_at": "2021-05-13T09:46:44+00:00",
>             "email": "email@example.com",
>             "external_customer_id": {
>                 "ecommerce": "5213433266347"
>             },
>             "first_charge_processed_at": "2021-05-13T13:46:39+00:00",
>             "first_name": "Banana",
>             "has_payment_method_in_dunning": false,
>             "has_valid_payment_method": true,
>             "hash": "f99bd4a6877257af",
>             "last_name": "Green",
>             "phone": "+11111111111",
>             "subscriptions_active_count": 1,
>             "subscriptions_total_count": 1,
>             "tax_exempt": false,
>             "updated_at": "2023-05-13T04:16:36+00:00"
>         }
>     }
>     ...
> }
> ```
> 
> It let us transform a record, so we can put missing fields from `include.customer.<field>` to `customer.<field>`. There are also some disadvatages: record will contain more data and we will have some duplicated fields. But advantages are: switching API version is a little bit destructive change because those versions are not backward compatible. So with this option we will break nothing but extend a model with some missing parts as well as keep API updated. I am not requesting change. This is just another option of how we can fix this problem

this is a good point, but the problem is we have fields in old api version that don't exist in a new one. so in future we can receive another oncall about missing fields.

**Comment by bazarnov** [comment]
> so in future we can receive another oncall about missing fields.

And sure we did: https://github.com/airbytehq/oncall/issues/2956

**Pull request merged**